### PR TITLE
chapter_13

### PR DESCRIPTION
**Pull request by muizzuddinyafiq**
ready

### CODE REVIEW

**Comment by muizzuddinyafiq** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  def correct_user
+    @micropost = current_user.microposts.find_by id: params[:id]
+    return if @micropost
+    flash[:danger] = "Micropost invalid"
+    redirect_to request.referrer || home_url
+  end
+end
+
```
Comment (ngocvt-0484) [inline app/controllers/microposts_controller.rb]: remove end line

File: app/helpers/microposts_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module MicropostsHelper
+end
```
Comment (ngocvt-0484) [inline app/helpers/microposts_helper.rb]: delete this file because not use

**Comment by muizzuddinyafiq** [comment]
ready

File: app/assets/stylesheets/microposts.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Microposts controller here.
```
Comment (ghost) [inline app/assets/stylesheets/microposts.scss]: delete this file please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by muizzuddinyafiq** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      flash[:success] = "Micropost created!"
+      redirect_to home_url
+    else
+      @feed_items = current_user.feed.page(params[:page])
+      render "static_pages/home"
+    end
+  end
+
+  def destroy
+  if @micropost.destroy
```
Comment (huongnt-2545) [inline app/controllers/microposts_controller.rb]: wrong indent
```suggestion
    if @micropost.destroy
```

File: test/controllers/microposts_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/microposts_controller_test.rb]: delete this file, it's not useful

File: test/fixtures/microposts.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/microposts.yml]: delete this file

File: test/models/micropost_test.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/models/micropost_test.rb]: delete this file

**Comment by muizzuddinyafiq** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged 
please merge this pull

**Pull request merged**